__docs__/wonder-blocks-form/checkbox.stories.tsx

```diff
@@ -88,6 +100,77 @@ Indeterminate.parameters = {
     },
 };
 
+export const IndeterminateWithGroup: StoryComponentType = () => {
+    const [allSelected, setAllSelected] = React.useState<boolean | null>(false);
+    const [selectedValues, setSelectedValues] = React.useState<Array<string>>(
+        [],
+    );
+    const choices = [
+        {label: "Pepperoni", value: "pepperoni"},
+        {label: "Sausage", value: "sausage"},
+        {label: "Extra cheese", value: "cheese"},
+        {label: "Green pepper", value: "pepper"},
+        {label: "Mushroom", value: "mushroom"},
+    ];
+
+    const handleSelectAll = () => {
+        if (allSelected || allSelected === null) {
+            setSelectedValues([]);
+            setAllSelected(false);
+        } else {
+            const allValues = choices.map((choice) => choice.value);
+            setSelectedValues(allValues);
+            setAllSelected(true);
+        }
+    };
+
+    const handleCheckboxGroupSelect = (values: Array<string>) => {
+        setSelectedValues(values);
+        if (values.length === choices.length) {
+            setAllSelected(true);
+        } else if (values.length) {
+            setAllSelected(null);
+        } else {
+            setAllSelected(false);
+        }
+    };
+
+    return (
+        <View>
+            <Checkbox
+                checked={allSelected}
+                label={"Topping(s)"}
+                onChange={handleSelectAll}
+            />
+            <Strut size={Spacing.small_12} />
+            <View style={{marginInlineStart: Spacing.large_24}}>
+                <CheckboxGroup
+                    groupName="toppings"
+                    onChange={handleCheckboxGroupSelect}
+                    selectedValues={selectedValues}
+                >
+                    {choices.map((choice) => (
+                        <Choice
+                            key={choice.label}
+                            label={choice.label}
+                            value={choice.value}
+                        />
+                    ))}
+                </CheckboxGroup>
+            </View>
+        </View>
+    );
+};
+
+IndeterminateWithGroup.parameters = {
+    docs: {
+        storyDescription: `Here is an example of how you can use the
+            indeterminate checkbox to select all or deselect all options
+            in a checkbox group. If only some of the options are selected,
+            the indeterminate checkbox will be in the indeterminate state.`,
+    },
+};
+
```

Comment:
I figured it would be useful to add an example of how someone might want to use indeterminate checkboxes in practice.